Controller/RecommendationController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * File containing the RecommendationController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RecommendationBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use EzSystems\RecommendationBundle\Client\YooChooseNotifier;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+class RecommendationController extends Controller
+{
+    public function recommendationsAction( Request $request )
+    {
+        if ( !$request->isXmlHttpRequest() ) {
+            throw new BadRequestHttpException();
+        }
+
+        $userId = $this->getRepository()->getCurrentUser()->id;
```

Comment:
Route arguments as in route/{userId}/{scenarioId}/{limit}, with named arguments in the Controller's method.

Do you expect very different arguments for this call ?